app/scripts/views/components/barChart/barChart.js

```diff
@@ -44,12 +45,15 @@
             } else if (d[data_.attrId] > opts_.xDomain[opts_.xDomain.length - 2] && !data_.hasNA) {
               val = opts_.xDomain[opts_.xDomain.length - 1];
             } else {
-              for (i = 2; i < opts_.xDomain.length - 2; i++) {
+              if (data_.hasNA) {
+                endNumIndex = opts_.xDomain.length - 2;
+              } 
+              for (i = 2; i < endNumIndex; i++) {
                 if (d[data_.attrId] <= opts_.xDomain[i] &&
                   d[data_.attrId] >= opts_.xDomain[i - 1]) {
                   _min = opts_.xDomain[i - 1];
                   _max = opts_.xDomain[i];
-                  val = _min + (_max - _min) / 3;
+                  val = _min + (_max - _min) / 4;
```

Comment:
I change it because I make gutter smaller when data is small number, which leads to bars become continuous. So It should be divided by 4 to separate bars.